File: Projects/FlutterPerfLab/lib/models/map_point.dart
import 'package:google_maps_flutter/google_maps_flutter.dart';

/// A lightweight, immutable Data Transfer Object (DTO) representing a distinct geographical annotation.
///
/// Enforcing strict immutability guarantees thread-safe compliance when transferring
/// spatial datasets across isolates.
class MapPoint {
  /// A deterministic integer sequence uniquely identifying the spatial coordinate.
  ///
  /// Bypassing UUID generation eliminates cryptographic CPU overhead. This integer
  /// acts as a unique identifier for UI elements, enabling efficient deterministic diffing
  /// and preventing full-screen repaints during rapid updates.
  final int id;

  /// The precise spatial coordinates (latitude and longitude) for map rendering.
  final LatLng coordinate;

  /// A descriptive label associated with the geographic coordinate.
  final String title;

  const MapPoint({
    required this.id,
    required this.coordinate,
    required this.title,
  });
}

File: Projects/NativeKotlinPerfLab/app/src/main/java/com/tez/nativekotlinperflabapp/models/MapPoint.kt
package com.tez.nativekotlinperflabapp.models

import com.google.android.gms.maps.model.LatLng

/**
 * A lightweight, thread-safe Data Transfer Object (DTO) representing a geographical map point.
 */
data class MapPoint(

    /**
     * A unique identifier required by Jetpack Compose for efficient list diffing
     * and stable rendering.
     */
    val id: Int,

    /**
     * The precise spatial coordinates (latitude and longitude) of the map point.
     */
    val coordinate: LatLng,

    /**
     * A localized, descriptive label associated with the geographic coordinate.
     */
    val title: String
)